TS.fsx

```diff
@@ -531,8 +531,25 @@ let EmitDictionaries flavor =
         | "Object" -> Pt.printl "interface %s {" dict.Name
         | _ -> Pt.printl "interface %s extends %s {" dict.Name dict.Extends
 
+        let emitJsonProperty (p: ItemsType.Root) =
+            Pt.printl "%s: %s;" p.Name.Value p.Type.Value
+
+        let removedPropNames = 
+            getRemovedItems ItemKind.Property flavor 
+            |> Array.filter (matchInterface dict.Name)
+            |> Array.map (fun rp -> rp.Name.Value)
+        let addedProps = 
+            getAddedItems ItemKind.Property flavor 
+            |> Array.filter (matchInterface dict.Name)
+
         Pt.increaseIndent()
-        dict.Members |> Array.iter (fun m -> Pt.printl "%s?: %s;" m.Name (DomTypeToTsType m.Type))
+        Array.iter emitJsonProperty addedProps
+        dict.Members
+        |> Array.filter (fun m -> not (Array.contains m.Name removedPropNames))
+        |> Array.iter (fun m -> 
+            match (findOverriddenItem m.Name ItemKind.Property dict.Name) with
+            | Some om -> emitJsonProperty om
+            | None -> Pt.printl "%s?: %s;" m.Name (DomTypeToTsType m.Type))
```

Comment:
there's always multiple way to do things in F#. My personal style preference would be to do
```fs
let isRemovedPropName =
    let removedPropNames = ...
    not << fun m -> Set.contains m.Name removedPropNames
(* ... find and emit addedProps ... *)
for m in Array.filter isRemovedPropName dict.Members do
  (* ... same match body as before ... *)
```

This makes the side-effecting syntactically obvious and reduces the overhead from F#'s lambda syntax.